types/internals/react.d.ts

```diff
@@ -26,4 +26,12 @@ export interface NextAuthConfig {
   _getSession: any
 }
 
-export type SessionContext = React.Context<Session>
+export type SessionContextValue<R extends boolean = false> = R extends true
+  ?
+      | { data: Session; status: "authenticated"; loading: false }
+      | { data: null; status: "loading"; loading: true }
+  :
+      | { data: Session; status: "authenticated"; loading: false }
+      | { data: null; status: "unauthenticated" | "loading"; loading: boolean }
```

Comment:
Let's have a call to discuss this one, I'm a bit confused about why the ternary here 🧐

Reply:
This is to enforce stricter types when using `required: true`